### PR TITLE
⚠️ Move InstanceID from Spec to Status

### PR DESCRIPTION
**Pull request by mdbooth**
This is a status field and should not be included in the spec. ProviderID is the same, but cannot be removed as it is required by Cluster API.

This PR replaces https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1961 with a smaller scope.

TODO:
- [ ] Integrate `restorev1alpha6MachineTemplateMachineSpec` into https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1989

/hold

### CODE REVIEW

File: api/v1beta1/openstackmachine_types.go [hunk 1]
```diff
@@ -119,6 +118,10 @@ type OpenStackMachineStatus struct {
 	// +optional
 	Ready bool `json:"ready"`
 
+	// InstanceID is the OpenStack instance ID for this machine.
+	// +optional
+	InstanceID optional.String `json:"instanceID,omitempty"`
```
Comment (MaysaMacedo) [inline api/v1beta1/openstackmachine_types.go:123]: Have we considered adding it under the `Resolved` field?

**Review by MaysaMacedo — COMMENTED** [review]
It lgtm, I just have one question:

File: api/v1beta1/openstackmachine_types.go (hunk 1 above)
Comment (mdbooth) [reply api/v1beta1/openstackmachine_types.go:123]: My first PR put it in `Resources`. This made sense, but I've decided against that for now. The reason is I think InstanceID is `fundamental` to the OpenStackMachine object and something an end user might be interested in (e.g. in `printcolumns`), but the other properties are not.

I have an idea that I might move actual machine creation into a separate controller with its own CRD, in which case `Resources` would move to the new controller. In that case I think we'd still want InstanceID to be on *this* object, so I decided to put it here instead.

None of which precludes adding it to Resources in the future too if it turns out that still makes sense.

**Pull request merged**